crates/trigger/src/cli.rs

```diff
@@ -164,13 +165,28 @@ where
         let mut builder = TriggerExecutorBuilder::new(loader);
         self.update_wasmtime_config(builder.wasmtime_config_mut())?;
 
-        let logging_hooks =
-            StdioLoggingTriggerHooks::new(self.follow_components(), self.log.clone());
+        let logging_hooks = StdioLoggingTriggerHooks::new(self.follow_components(), self.log_dir());
         builder.hooks(logging_hooks);
 
         builder.build(locked_url, trigger_config).await
     }
 
+    pub fn log_dir(&self) -> Option<PathBuf> {
+        match &self.log {
+            Some(l) => return Some(l.clone()),
+            None => {
+                if self.from_registry {
```

Comment:
I know this is a decreasing consideration, but `from_registry` will be _false_ for bindles.

Do we have access to the `origin` attribute here?  If so that would be safer I think.

Reply:
We could check if the SPIN_STATE_DIR is set and if it is not set, return None.

Reply:
updated.